File: fixtures/python/get_complex_url_params.py
import requests

params = (
    ('page', '1'),
    ('available', ['', '1']),
    ('location', '0'),
    ('city[id]', '0'),
    ('city[locality]', ''),
    ('city[locality_text]', ''),
    ('city[administrative_area_level_2]', ''),
    ('city[administrative_area_level_2_text]', ''),
    ('city[administrative_area_level_1]', ''),
    ('city[administrative_area_level_1_text]', ''),
    ('city[country]', ''),
    ('city[country_text]', ''),
    ('city[latitude]', ''),
    ('city[longitude]', ''),
    ('city[zoom]', ''),
    ('city[name]', ''),
    ('region[id]', '0'),
    ('region[locality]', ''),
    ('region[locality_text]', ''),
    ('region[administrative_area_level_2]', ''),
    ('region[administrative_area_level_2_text]', ''),
    ('region[administrative_area_level_1]', ''),
    ('region[administrative_area_level_1_text]', ''),
    ('region[country]', ''),
    ('region[country_text]', ''),
    ('region[latitude]', ''),
    ('region[longitude]', ''),
    ('region[zoom]', ''),
    ('region[name]', ''),
    ('country', ''),
    ('environment', ''),
    ('population', ''),
    ('period', '0'),
    ('date', '2017-03-03'),
    ('datestart', '2017-03-03'),
    ('dateend', '2017-06-24'),
    ('season', ''),
    ('duration', ''),
    ('isfd', ''),
    ('stopover', ''),
)

response = requests.get('https://www.nomador.com/house-sitting/', params=params)

#NB. Original query string below. It seems impossible to parse and
#reproduce query strings 100% accurately so the one below is given
#in case the reproduced version is not "correct".
# response = requests.get('https://www.nomador.com/house-sitting/?page=1&available=&available=1&location=0&city%5Bid%5D=0&city%5Blocality%5D=&city%5Blocality_text%5D=&city%5Badministrative_area_level_2%5D=&city%5Badministrative_area_level_2_text%5D=&city%5Badministrative_area_level_1%5D=&city%5Badministrative_area_level_1_text%5D=&city%5Bcountry%5D=&city%5Bcountry_text%5D=&city%5Blatitude%5D=&city%5Blongitude%5D=&city%5Bzoom%5D=&city%5Bname%5D=&region%5Bid%5D=0&region%5Blocality%5D=&region%5Blocality_text%5D=&region%5Badministrative_area_level_2%5D=&region%5Badministrative_area_level_2_text%5D=&region%5Badministrative_area_level_1%5D=&region%5Badministrative_area_level_1_text%5D=&region%5Bcountry%5D=&region%5Bcountry_text%5D=&region%5Blatitude%5D=&region%5Blongitude%5D=&region%5Bzoom%5D=&region%5Bname%5D=&country=&environment=&population=&period=0&date=2017-03-03&datestart=2017-03-03&dateend=2017-06-24&season=&duration=&isfd=&stopover=')

File: test/fixtures/node-axios/get_complex_url_params.js
import axios from 'axios';

const params = new URLSearchParams();
params.append('page', '1');
params.append('available', '');
params.append('available', '1');
params.append('location', '0');
params.append('city[id]', '0');
params.append('city[locality]', '');
params.append('city[locality_text]', '');
params.append('city[administrative_area_level_2]', '');
params.append('city[administrative_area_level_2_text]', '');
params.append('city[administrative_area_level_1]', '');
params.append('city[administrative_area_level_1_text]', '');
params.append('city[country]', '');
params.append('city[country_text]', '');
params.append('city[latitude]', '');
params.append('city[longitude]', '');
params.append('city[zoom]', '');
params.append('city[name]', '');
params.append('region[id]', '0');
params.append('region[locality]', '');
params.append('region[locality_text]', '');
params.append('region[administrative_area_level_2]', '');
params.append('region[administrative_area_level_2_text]', '');
params.append('region[administrative_area_level_1]', '');
params.append('region[administrative_area_level_1_text]', '');
params.append('region[country]', '');
params.append('region[country_text]', '');
params.append('region[latitude]', '');
params.append('region[longitude]', '');
params.append('region[zoom]', '');
params.append('region[name]', '');
params.append('country', '');
params.append('environment', '');
params.append('population', '');
params.append('period', '0');
params.append('date', '2017-03-03');
params.append('datestart', '2017-03-03');
params.append('dateend', '2017-06-24');
params.append('season', '');
params.append('duration', '');
params.append('isfd', '');
params.append('stopover', '');

const response = await axios.get('http://localhost:28139/house-sitting/', {
  params: params
});
